compiler/semfold.nim

```diff
@@ -268,6 +268,17 @@ proc evalOp(m: TMagic, n, a, b, c: PNode; g: ModuleGraph): PNode =
     of tyInt64, tyInt, tyUInt..tyUInt64:
       result = newIntNodeT(`shr`(getInt(a), getInt(b)), n, g)
     else: internalError(g.config, n.info, "constant folding for shr")
+  of mAshrI:
+    when defined(nimAshr):
```

Comment:
This is not required, only the stdlib needs ``when defined(nimAshr)`` protection, the compiler knows about ``mAshrI`` already because it's part of its enum.

Reply:
but why I failed to bootstrap the compiler using prev version(before ashr added)?